### ISSUE TITLE
Need dependency list to fix where dependencies 'were found, but they required a higher minimum deployment target'

### ISSUE THREAD

**Issue by TCOA** [mapbox/mapbox-navigation-ios#206]
I have been using Mapbox to build my app and have bits/pieces working - then, today, I found this code (yeah, as it contains many things I was worried about how to include!)

my podfile (working) looks like this:

pod 'MapboxCoreNavigation', '~> 0.3.0'
pod 'Mapbox-iOS-SDK', '~> 3.5'
pod 'MapboxDirections.swift', '~> 0.9.0'
pod 'MapboxStatic.swift', '~> 0.8'

Yet, when I try to add
pod 'MapboxNavigation', '~> 0.3.0'
    
I get
`Analyzing dependencies
[!] Unable to satisfy the following requirements:

- `MapboxNavigation (~> 0.3.0)` required by `Podfile`

Specs satisfying the `MapboxNavigation (~> 0.3.0)` dependency were found, but they required a higher minimum deployment target.`    

I tried doing a pod upgrade on each working 'Mapbox....' piece, a repo update and even 'sudo gem install cocoapods'  (all the upgrade things I could find/think of...) - still get the error......

Any help/clarification on just what else to do will be greatly appreciated.

**Comment by bsudekum** [comment]
cc @1ec5

**Comment by 1ec5** [comment]
What is your Podfile’s minimum deployment target?

MapboxNavigation [requires iOS 9.0](https://github.com/mapbox/mapbox-navigation-ios/blob/5be6ada250e43804d3d74c5724687a761abb6271/MapboxNavigation.podspec#L26) as the minimum deployment target. MapboxCoreNavigation [requires iOS 8.0](https://github.com/mapbox/mapbox-navigation-ios/blob/5be6ada250e43804d3d74c5724687a761abb6271/MapboxCoreNavigation.podspec#L26), as do other dependencies like [OSRMTextInstructions](https://github.com/Project-OSRM/osrm-text-instructions.swift/blob/03dd2b8d597c51d74c958874eb4edbb0b9e2cdf3/OSRMTextInstructions.podspec#L27) and [MapboxDirections.swift](https://github.com/mapbox/MapboxDirections.swift/blob/4c8f3f9a825979018f8d06b3771b9dc7c55eb4e3/MapboxDirections.swift.podspec#L27).

**Comment by TCOA** [comment]
Thanks for the info.

I was targeting 8.0 (requested by the users), though MapboxNavigation is a valuable enough addition to force things to 9.0.

Changing that in my podfile, I got this from pod install....

`Analyzing dependencies
Downloading dependencies
Installing AWSCore (2.5.5)
Installing AWSPolly (2.5.5)
Using Alamofire (4.4.0)
Using Bolts (1.8.4)
Using Eureka (2.0.1)
Using FBSDKCoreKit (4.20.2)
Using FBSDKLoginKit (4.20.2)
Using Firebase (3.15.0)
Using FirebaseAnalytics (3.7.0)
Using FirebaseAuth (3.1.1)
Using FirebaseCore (3.5.2)
Using FirebaseInstanceID (1.0.9)
Using FirebaseMessaging (1.2.2)
Using Font-Awesome-Swift (1.6.1)
Using GTMOAuth2 (1.1.4)
Using GTMSessionFetcher (1.1.9)
Using GoogleSignIn (4.0.2)
Using GoogleToolboxForMac (2.1.1)
Using ImageRow (1.0.0)
Using Mapbox-iOS-SDK (3.5.0)
Using MapboxCoreNavigation (0.3.0)
Using MapboxDirections.swift (0.9.0)
Installing MapboxNavigation (0.3.0)
Using MapboxStatic.swift (0.8.0)
Using OSRMTextInstructions (0.1.1)
Using PDFReader (2.3.1)
Using Polyline (4.1.1)
Using Protobuf (3.2.0)
Installing Pulley (1.4)
Installing SDWebImage (4.0.0)
Using SwiftKeychainWrapper (3.0.1)
Using SwiftyJSON (3.1.4)
Generating Pods project
Integrating client project
Sending stats
Pod installation complete! There are 17 dependencies from the Podfile and 32 total pods installed.

[!] Smart quotes were detected and ignored in your Podfile. To avoid issues in the future, you should not use TextEdit for editing it. If you are not using TextEdit, you should turn off smart quotes in your editor of choice.

[!] Can't merge user_target_xcconfig for pod targets: ["OSRMTextInstructions", "Pulley"]. Singular build setting SWIFT_VERSION has different values.

[!] Can't merge user_target_xcconfig for pod targets: ["OSRMTextInstructions", "Pulley"]. Singular build setting SWIFT_VERSION has different values.

[!] The `App [Debug]` target overrides the `ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES` build setting defined in `Pods/Target Support Files/Pods-App/Pods-App.debug.xcconfig'. This can lead to problems with the CocoaPods installation
    - Use the `$(inherited)` flag, or
    - Remove the build settings from the target.
`
I'm not sure if the notices might affect things (this is a first Xcode app) nor how to fix them, exactly (I tried setting ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = NO, but it didn't help)

when trying to compile, I get other errors now (no token from Firebase, which is very odd), so I have to figure that out before I can get back to the map as I can't even log in now......., though it appears  (on the surface anyway) that going to 9.0 in the pod at least allowed the install to go through.

**Comment by bsudekum** [comment]
Noting the Firebase bug is unrelated to this library. 

Regarding `ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES `, can you try https://github.com/CocoaPods/CocoaPods/issues/5981#issuecomment-269548468?

**Comment by TCOA** [comment]
Confirming that the Firebase bug certainly was not related (I knew that, though had to work around it to get back on getting Mapbox stuff to work).

Also confirming that 'unbolding' ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES and re-running pod install worked to remove that error.

Any ideas on the "Pulley" thing?  (the Smart quotes thing has always been there and probably because I used TextEdit to edit it {just got this Mac and it seemed logical to use Apple's editor to edit their files...... - obviously that wasn't the right thing to do....} - I will get it solved later)

I've fixed the Firebase issue (token refresh wasn't properly set) and now starting to work with the map again.

To make sure I am doing things right, I downloaded and tried the example project (https://github.com/mapbox/mapbox-navigation-ios#running-the-example-project), though in trying to follow the instructions, I can't as I don't have/use carthage (#2 says "Run carthage update --platform ios to build just the iOS dependencies") - are there other instructions?  (I've tried various guesses, but no success so far.......  I will open another ticket on this if you prefer, though it would be great if you can update the main page with step-by-step to work without having to install carthage (as you can tell, I'm having enough trouble just getting pods to work! :)

**Comment by bsudekum** [comment]
The pulley warnings should be fixed with https://github.com/Project-OSRM/osrm-text-instructions.swift/commit/03dd2b8d597c51d74c958874eb4edbb0b9e2cdf3

**Comment by 1ec5** [comment]
Once Project-OSRM/osrm-text-instructions.swift#23 lands, we’ll release a v0.1.2 of that library so that a `pod update` will automatically pull down the fix.

**Comment by 1ec5** [comment]
Please try `pod update` and let us know if the problem persists. Thanks!

**Comment by TCOA** [comment]
"Pulley" problem went away with pod update (yeah!).

Only thing now is the 
`[!] The `App [Release]` target overrides the `SWIFT_VERSION` build setting defined in `Pods/Target Support Files/Pods-App/Pods-App.release.xcconfig'. This can lead to problems with the CocoaPods installation
    - Use the `$(inherited)` flag, or
    - Remove the build settings from the target.
`

I tried the things from CocoaPods/CocoaPods#5981, which fixed the one problem (above), though this one is still around.....  It doesn't look like it is critical, and now with all the pod updates I have to go through and test all my other features (and somewhere along the line of all this I have lost remote notifications which is critical for using the Navigation map, so now I have to go fix that first.....!)

Thanks for the fast response on the other issues - I'll keep testing and report anything else concrete that I get into.

**Comment by 1ec5** [comment]
Regarding the `SWIFT_VERSION` build setting warning, this could be a remnant of Project-OSRM/osrm-text-instructions.swift#22. If you unset (unbold) that setting manually, it hopefully won’t get set again the next time you install or update pods.

**Comment by TCOA** [comment]
After several attempts, and unsetting (unbold) not only the "Always Embed Swift Standard Libraries" setting, but all 'bold' settings (others reading this - be observant on what the setting was before hitting that 'delete' button - some bold entries need to be re-entered for your app to work!), then running pod install again - no more error message!
